types/index.d.ts

```diff
@@ -146,7 +147,7 @@ declare module 'mongoose' {
     /**
      * Create a new schema
      */
-    constructor(definition?: SchemaDefinition<SchemaDefinitionType<EnforcedDocType>> | DocType, options?: SchemaOptions<TPathTypeKey, DocType, TInstanceMethods, TQueryHelpers, TStaticMethods>);
+    constructor(definition?: SchemaDefinition<SchemaDefinitionType<EnforcedDocType>> | DocType, options?: SchemaOptions<TPathTypeKey, DocType, TInstanceMethods, TQueryHelpers, TVirtuals, TStaticMethods>);
```

Comment:
Can we add `TVirtuals` as the last generic param, instead of pushing `TStaticMethods` back one? Backwards compat.

Reply:
Since TStaticMethods  has been added by auto-typed-schema  PR and that will be available in 6.4 and hopefully this PR as well, so It will not be a problem. I would like to keep the same order as schema generics.
After that said, What do you think?

Reply:
I suppose that's fine as long as we don't expect people to instantiate `SchemaOptions` themselves. Which is a sufficiently rare case, I think.

Reply:
I will apply the desired changes because 6.4 is already merged :)

Reply:
done